python/dask_cudf/dask_cudf/backends.py

```diff
@@ -446,91 +450,117 @@ def _default_backend(func, *args, **kwargs):
-            )
+    @to_pandas_dispatch.register((cudf.DataFrame, cudf.Series, cudf.Index))
+    def to_pandas_dispatch_from_cudf(data, **kwargs):
+        return data.to_pandas(**kwargs)
 
-        @staticmethod
-        def read_parquet(*args, engine=None, **kwargs):
-            from dask_cudf.io.parquet import CudfEngine
 
-            return _default_backend(
-                dd.read_parquet,
-                *args,
-                engine=CudfEngine,
-                **kwargs,
-            )
+to_cudf_dispatch = Dispatch("to_cudf_dispatch")
 
-        @staticmethod
-        def read_json(*args, **kwargs):
-            from dask_cudf.io.json import read_json
 
-            return read_json(*args, **kwargs)
+@to_cudf_dispatch.register((pd.DataFrame, pd.Series, pd.Index))
+def to_cudf_dispatch_from_pandas(data, **kwargs):
+    return cudf.from_pandas(data, **kwargs)
 
-        @staticmethod
-        def read_orc(*args, **kwargs):
-            from dask_cudf.io import read_orc
 
-            return read_orc(*args, **kwargs)
+@to_cudf_dispatch.register((cudf.DataFrame, cudf.Series, cudf.Index))
+def to_cudf_dispatch_from_cudf(data, **kwargs):
+    return data
 
-        @staticmethod
-        def read_csv(*args, **kwargs):
-            from dask_cudf.io import read_csv
 
-            return read_csv(*args, **kwargs)
+# Define "cudf" backend engine to be registered with Dask
+class CudfBackendEntrypoint(DataFrameBackendEntrypoint):
+    """Backend-entrypoint class for Dask-DataFrame
 
-        @staticmethod
-        def read_hdf(*args, **kwargs):
-            from dask_cudf import from_dask_dataframe
+    This class is registered under the name "cudf" for the
+    ``dask.dataframe.backends`` entrypoint in ``setup.cfg``.
+    Dask-DataFrame will use the methods defined in this class
+    in place of ``dask.dataframe.<creation-method>`` when the
+    "dataframe.backend" configuration is set to "cudf":
 
-            # HDF5 reader not yet implemented in cudf
-            warnings.warn(
-                "read_hdf is not yet implemented in cudf/dask_cudf. "
-                "Moving to cudf from pandas. Expect poor performance!"
-            )
-            return from_dask_dataframe(
-                _default_backend(dd.read_hdf, *args, **kwargs)
-            )
+    Examples
+    --------
+    >>> import dask
+    >>> import dask.dataframe as dd
+    >>> with dask.config.set({"dataframe.backend": "cudf"}):
+    ...     ddf = dd.from_dict({"a": range(10)})
+    >>> type(ddf)
+    <class 'dask_cudf.core.DataFrame'>
+    """
 
-except ImportError:
-    pass
+    @classmethod
+    def to_backend_dispatch(cls):
+        return to_cudf_dispatch
+
+    @classmethod
+    def to_backend(cls, data: dd.core._Frame, **kwargs):
+        if isinstance(data._meta, (cudf.DataFrame, cudf.Series, cudf.Index)):
+            # Already a cudf-backed collection
+            return data
+        return data.map_partitions(cls.to_backend_dispatch(), **kwargs)
```

Comment:
Good question - I updated things a bit to be explicit about supported/unsupported key-word arguments.  I also added a bit more test coverage.

Context: We (probably) **do** want to make it possible to do things like `ddf.to_backend("cudf", nan_as_null=False)`.  However, I'm still unsure of the best way to *document* the supported `to_backend` kwargs for the various conversions.

Reply:
Hmmm, can we link to the registered backend dispatch function which could explain what it accepts as keyword arguments?

Reply:
I’ll give this a think, but not sure how/where to link to the dispatch-function args, since the available arguments to the user-facing API (to_backend) will depend on the specified backend. Any ideas?